src/parser/jobs/mnk/modules/DKOptimalGoof.tsx

```diff
@@ -0,0 +1,202 @@
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {Actors} from 'parser/core/modules/Actors'
+import Checklist, {Requirement, Rule} from 'parser/core/modules/Checklist'
+import {Data} from 'parser/core/modules/Data'
+import {Timeline} from 'parser/core/modules/Timeline'
+import React from 'react'
+import {Button, Table} from 'semantic-ui-react'
+import {DISPLAY_ORDER} from './DISPLAY_ORDER'
+
+interface GCD {
+	timestamp: number
+	action: Action
+	dfTimer?: number
+	reason?: string
+}
+
+const REFRESH_TWIN_WINDOW: number = 3000
+
+export class DKOptimalGoof extends Analyser {
+	static override handle = 'DragonKickRotation'
+
+	@dependency private checklist!: Checklist
+
+	@dependency private data!: Data
+	@dependency private timeline!: Timeline
+	@dependency private actors!: Actors
+
+	private readonly blitz = [
+		this.data.actions.CELESTIAL_REVOLUTION.id,
+		this.data.actions.ELIXIR_FIELD.id,
+		this.data.actions.RISING_PHOENIX.id,
+		this.data.actions.PHANTOM_RUSH.id,
+	]
+
+	private dragonKicks: GCD[] = []
+	private missedKicks: GCD[] = []
+	private twinRefreshExpected?: number
+
+	override initialise(): void {
+		if (getIsAprilFirst()) {
+			const playerFilter = filter<Event>().source(this.parser.actor.id)
+
+			this.addEventHook(playerFilter.type('action'), this.onCast)
+			this.addEventHook(playerFilter.type('statusApply').status(this.data.statuses.DISCIPLINED_FIST.id), this.onGain)
+			this.addEventHook(playerFilter.type('statusRemove').status(this.data.statuses.DISCIPLINED_FIST.id), this.onDrop)
+
+			this.addEventHook('complete', this.onComplete)
+		}
+	}
+
+	private onCast(event: Events['action']) {
+		const action = this.data.getAction(event.action)
+
+		const inPB = this.actors.current.hasStatus(this.data.statuses.PERFECT_BALANCE.id)
+		if (action == null || !(action.onGcd ?? false) || inPB  || this.blitz.includes(action.id)) { return }
+
+		const gcd: GCD = {
+			timestamp: event.timestamp,
+			action: action,
+		}
+
+		const isDragonKick = action.id === this.data.actions.DRAGON_KICK.id
+		const inDisciplinedFist = this.actors.current.hasStatus(this.data.statuses.DISCIPLINED_FIST.id)
+		let needTwinRefresh = true
+		if (this.twinRefreshExpected) {
+			needTwinRefresh = this.twinRefreshExpected - event.timestamp < REFRESH_TWIN_WINDOW
+			gcd.dfTimer = this.twinRefreshExpected  - event.timestamp
+		}
+
+		if (this.actors.current.hasStatus(this.data.statuses.LEADEN_FIST.id) && (
+			this.actors.current.hasStatus(this.data.statuses.OPO_OPO_FORM.id)
+				|| this.actors.current.hasStatus(this.data.statuses.FORMLESS_FIST.id))) {
+			return
+		}
+
+		if (!inDisciplinedFist || this.actors.current.hasStatus(this.data.statuses.COEURL_FORM.id)) { return }
+
+		if (!needTwinRefresh) {
+			if (isDragonKick) {
+				this.dragonKicks.push(gcd)
+			} else {
+				gcd.reason = 'Not a dragon kick :('
+				this.missedKicks.push(gcd)
+			}
+		}
+	}
+
+	onComplete() {
+		this.checklist.add(new Rule({
```

Comment:
My one concern with putting this in the checklist is that we might accidentally get a newer player thinking they really should be doing this! I'd be more comfortable putting this as a SEVERITY.MEMES Suggestion, since that way it's pretty clear that we're not serious about it. Otherwise, hilarious!

Reply:
Understood, that makes sense, and it will give me a chance to play with suggestions! I'll rework to use suggestions instead. Do you think the output table is fine?  Is there something I should add there to make it clear its a meme?

Reply:
I think the 'not a dragon kick (frown)' should be silly enough to be a hint that it's not real!

Reply:
Removed the checklist item and added a suggestion 
![image](https://github.com/xivanalysis/xivanalysis/assets/59540783/0ac4787f-bb51-4d93-9902-4bba37dfa0a5)